rslib/src/search/parser.rs

```diff
@@ -400,6 +415,63 @@ fn parse_prop(s: &str) -> ParseResult<SearchNode> {
                 FailKind::InvalidPropInteger(format!("{}{}", prop, operator)),
             ));
         }
+    } else if prop == "rated" {
+        let mut it = num.splitn(2, ':');
+
+        let days: i32 = if let Ok(i) = it.next().unwrap().parse::<i32>() {
+            i
+        } else {
+            return Err(parse_failure(
+                s,
+                FailKind::InvalidPropInteger(format!("{}{}", prop, operator)),
+            ));
+        };
+
+        let (newop, days) = if days > 0 {
+            flip_operators(&operator, days)
+        } else {
+            (&*operator, days)
+        };
+        operator = newop.to_string();
+
+        let ease = match it.next() {
+            Some(v) => {
+                if let Ok(u) = v.parse::<u8>() {
+                    if (1..5).contains(&u) {
+                        EaseKind::AnswerButton(u)
+                    } else {
+                        return Err(parse_failure(
+                            s,
+                            FailKind::InvalidPropInteger(format!("{}{}", prop, operator)),
```

Comment:
`InvalidPropInteger` does not really apply here. The generic `search-invalid-argument` is available, but since this will be a new feature I think it would be preferrable to provide some information about the expected value.
`search-invalid-rated-ease` could be easily generalised to work for both `rated:` and `prop:rated`.
@dae, seeing this I realise that I could and probably should have made the error strings more generic. There is no need to bind `search-invalid-prop-float` to a property search, for example. You haven't pushed them to Pontoon, yet, have you? Then I'll make a quick PR as soon as #926 is merged.

Reply:
I added `search-invalid-rated-ease` within this PR now.